src/commands.ts

```diff
@@ -0,0 +1,222 @@
+// -----------------------------------------------------------------------------
+// Utilities for command parsing
+// See chesster.js for command definitions.
+// -----------------------------------------------------------------------------
+import _ from 'lodash'
+
+const VARIABLE = /^\{(int)(text)|:[a-zA-Z]+\}$/
+const CHOICE_VARIABLE = /^\{([a-zA-Z0-9]+\|?)+\}$/
+const CONSTANT_VALUE = /^[a-zA-Z]+$/
+
+export class TextVariable {
+    typeName = 'text'
+    value: string
+    constructor(token: string) {
+        this.value = token
+    }
+}
+
+export class IntVariable {
+    typeName = 'int'
+    value: number
+    constructor(token: string) {
+        this.value = parseInt(token, 10)
+        if (isNaN(this.value)) throw new InvalidTypeValueError(token, 'int')
+    }
+}
+
+type Variable = TextVariable | IntVariable
+type VariableConstructor = new (token: string) => Variable
+const VARIABLE_TYPES: Record<string, VariableConstructor> = {
+    text: TextVariable,
+    int: IntVariable,
+}
+export function isText(v: Variable): v is TextVariable {
```

Comment:
To verify (since I'm not familiar with this construction), this means that a true return value implies the type of v is TextVariable?

Reply:
Yes, it's called a 'type guard' in typescript and it allows the compiler to know that the variable is that type from then on out